packages/alea-frontend/components/LoRelations.tsx

```diff
@@ -0,0 +1,173 @@
+        .filter((data: string) => data.startsWith('http://mathhub.info/ulo#cognitive-dimension='))
+        .map((data: string) => {
+          const encodedValue = data.split('=')[1];
+          return decodeURIComponent(encodedValue);
+        });
+      const crossRefs = relatedData
+        .filter((data: string) => data.startsWith('http://mathhub.info/ulo#crossrefs='))
+        .map((data: string) => {
+          const encodedValue = data.split('=')[1];
+          const decodedValue = decodeURIComponent(encodedValue);
+          return decodedValue.endsWith('#') ? decodedValue.slice(0, -1) : decodedValue;
+        });
+      return {
+        learningObject: binding.learningObject.value,
+        obj1: binding.obj1.value,
+        cognitiveDimensions,
+        crossRefs,
+      };
+    });
+    const finalString = transformedData
+      .flatMap(({ cognitiveDimensions, crossRefs }) =>
+        cognitiveDimensions.flatMap((dim) =>
+          crossRefs.map((ref) => `${dim}:${encodeURIComponent(ref)}`)
+        )
+      )
+      .join(',');
+    return finalString;
+  } catch (error) {
+    console.error('Error processing data:', error);
+  }
+}
+
+function processConceptOnlyData(result) {
+  try {
+    const groupedData = result.results.bindings.reduce((acc, { learningObject, obj1 }) => {
+      const learningObjectValue = learningObject.value;
+      const obj1Value = obj1.value;
+      if (!acc[learningObjectValue]) {
+        acc[learningObjectValue] = [];
+      }
+      acc[learningObjectValue].push(obj1Value);
+      return acc;
+    }, {});
+    const finalString = Object.values(groupedData).flat().join(',');
+    return finalString;
+  } catch (error) {
+    console.error('Error processing data:', error);
+  }
+}
+
+const LoRelations = ({ uri }: { uri: string }) => {
+  const { mmtUrl } = useContext(ServerLinksContext);
+  const [data, setData] = useState<Record<AllLoRelationTypes, string>>({
+    objective: '',
+    precondition: '',
+    crossrefs: '',
+    'example-for': '',
+    defines: '',
+    specifies: '',
+  });
+  useEffect(() => {
+    const processData = async () => {
+      try {
+        if (!uri?.length) return;
+        const updatedData = { ...data };
+        await Promise.all(
+          ALL_LO_RELATION_TYPES.map(async (value: AllLoRelationTypes) => {
+            const query = getSparqlQueryForLoRelation(uri, value);
+            const result = await sparqlQuery(mmtUrl, query);
+            const transformedData = ALL_DIM_CONCEPT_PAIR.includes(
+              value as LoRelationToDimAndConceptPair
+            )
+              ? processDimAndConceptData(result)
+              : processConceptOnlyData(result);
+            updatedData[value] = transformedData;
+          })
+        );
+        setData(updatedData);
+      } catch (error) {
+        console.error('Error processing data:', error);
+      }
+    };
+    processData();
+  }, [uri]);
+
+  const displayData = [
```

Comment:
There is no need for this this list. Get types for all  `AllLoRelationTypes`  and use `capitalizeFirstLetter` to get the title.
To fetch the relevant data, us `data[relation]`

Reply:
Made changes.